### ISSUE TITLE
`max-len` rule `ignoreTemplateLiterals` not ignore multi-line template literals

### ISSUE THREAD

**Issue by RanadeepPolavarapu** [eslint/eslint#7125]
**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** 6.2.1
* **npm Version:** 3.9.6

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-eslint

**Please show your full configuration:** Isolated config and used only one rule.
```yaml
        "max-len": [
            2,
            100, {
                "ignoreComments": true,
                "ignoreTrailingComments": true,
                "ignoreUrls": true,
                "ignoreStrings": true,
                "ignoreTemplateLiterals": true
            }
        ]
```

**What did you do? Please include the actual source code causing the issue.**
```js
        this.x = `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
        aaaaaaaaaaaaa`;
```
![untitled](https://cloud.githubusercontent.com/assets/7084995/18440503/625bcbe8-78d7-11e6-9b40-9f9c07c4134b.png)

**What did you expect to happen?** I expected eslint to **not** throw a `max-len` error as I have **ignoreTemplateLiterals** turned on.

**What actually happened? Please include the actual, raw output from ESLint.** It threw an error for multi-line template literal.

/cc #5805 and #7049.

**Comment by btmills** [comment]
Thanks for the report, @RanadeepPolavarapu! Easy repro in the [demo](http://eslint.org/demo):

```js
/* eslint max-len: ["error", 16, { "ignoreComments": true, "ignoreTemplateLiterals": true }] */

const a = `asdffdsaasdffdsaasdffdsa
asdffdsaasdffdsaasdffdsa
asdffdsaasdffdsaasdffdsa
asdffdsaasdffdsaasdffdsa`;
```

This gives a `max-len` error on the second and third lines of the literal. It looks like [`groupByLineNumber`](https://github.com/eslint/eslint/blob/c85fd84d627dc2788197b9cba26cb843dbe57d8d/lib/rules/max-len.js#L237-L241) is only handling the first and last lines of the literal and not any interior lines.

@RanadeepPolavarapu do you want to put together a pull request to fix this?

**Comment by platinumazure** [comment]
@ljharb FYI. Any chance you can take a look at this (and maybe help @RanadeepPolavarapu if s/he wants to put together a PR)? Thanks!

**Comment by ljharb** [comment]
Absolutely - @RanadeepPolavarapu please ping me if you need any help, or if you're not interested in making a PR, I can work on it.

**Comment by ljharb** [comment]
I'm going to put up a PR for this shortly, since this is my bug - @RanadeepPolavarapu, I hope you don't mind!

**Comment by ljharb** [comment]
https://github.com/eslint/eslint/pull/7138 filed

**Comment by RanadeepPolavarapu** [comment]
@ljharb Thanks for the PR. I apologize for my late reply, I fell ill and was resting.
